crates/shared/src/zeroex_api.rs

```diff
@@ -575,6 +575,87 @@ impl Metrics {
     }
 }
 
+pub mod websocket {
+    use super::*;
+
+    #[serde_as]
+    #[derive(Debug, Derivative, Clone, Deserialize, Eq, PartialEq)]
+    #[derivative(Default)]
+    #[serde(rename_all = "UPPERCASE")]
+    pub enum OrderState {
+        #[derivative(Default)]
+        Added,
+        Fillable,
+        Updated,
+        Expired,
+    }
+
+    #[serde_as]
+    #[derive(Debug, Derivative, Clone, Deserialize, Eq, PartialEq)]
+    #[derivative(Default)]
+    #[serde(rename_all = "camelCase")]
+    pub struct OrderMetadata {
+        #[serde(with = "model::bytes_hex")]
+        pub order_hash: Vec<u8>,
+        #[serde_as(as = "DisplayFromStr")]
+        pub remaining_fillable_taker_amount: u128,
+        pub state: OrderState,
+    }
+
+    #[derive(Debug, Default, Clone, Deserialize, Eq, PartialEq)]
+    pub struct OrderRecord {
+        #[serde(rename = "metaData")]
+        pub metadata: OrderMetadata,
+        pub order: Order,
+    }
+
+    impl OrderRecord {
+        /// Scales the `maker_amount` according to how much of the partially
+        /// fillable amount was already used.
+        pub fn remaining_maker_amount(&self) -> Result<u128> {
+            if self.metadata.remaining_fillable_taker_amount > self.order.taker_amount {
+                anyhow::bail!("remaining taker amount bigger than total taker amount");
+            }
+
+            // all numbers are at most u128::MAX so none of these operations can overflow
+            let scaled_maker_amount = U256::from(self.order.maker_amount)
+                * U256::from(self.metadata.remaining_fillable_taker_amount)
+                    .checked_div(self.order.taker_amount.into())
+                    .context("taker_amount is zero")?;
+
+            // `scaled_maker_amount` is at most as big as `maker_amount` which already fits
+            // in an u128
+            Ok(scaled_maker_amount.as_u128())
+        }
+    }
+
+    impl From<super::OrderRecord> for OrderRecord {
+        fn from(order_record: super::OrderRecord) -> Self {
+            Self {
+                metadata: OrderMetadata {
+                    order_hash: order_record.metadata.order_hash,
+                    remaining_fillable_taker_amount: order_record
+                        .metadata
+                        .remaining_fillable_taker_amount,
+                    state: OrderState::Added,
+                },
+                order: order_record.order,
+            }
+        }
+    }
+
+    /// A Ox API `orders` response.
+    #[derive(Debug, Default, Clone, Deserialize, Eq, PartialEq)]
+    #[serde(rename_all = "camelCase")]
+    pub struct OrdersResponse {
+        #[serde(rename = "type")]
+        pub _type: String,
```

Comment:
Using `_` on an identifier means the compiler will not complain if it's no longer used anywhere. Since we are already renaming the type we might as well find a name that does not require `_` as a prefix.
```suggestion
        pub kind: String,
```